plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -680,12 +684,32 @@ public void updateViewColumnComment(ConnectorSession session, SchemaTableName sc
         replaceViewVersion.commit();
     }
 
+    private String hashCredentials(Map<String, String> credentials)
+    {
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            String input = credentials.toString();
```

Comment:
Are you sure that Map has a properly-implemented `toString()` and that it's not just outputting the class name and some instance ID?

Reply:
I added a logging statement and gave it a test on udev. It's printing out the credentials as expected!

Reply:
Looks like this: {credential=<application-id>:<client-secret>}